scripts: compliance: More cleanups

carlescufi opened the pull request:
More cleanups in preparation for some extra annotation work.

carlescufi committed:
scripts: compliance: Remove unneeded workaround

The workaround for old pytest bug
https://github.com/PyCQA/pylint/issues/2906
is not required anymore after more than 3 years, so remove it.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
scripts: compliance: Consolidate 'git diff' calls in get_files()

Many tests invoke `git diff` independently with very similar parameters
to obtain a filtered list of files.
Consolidate those into a new `get_files()` function.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
scripts: compliance: Remove old TODO

This TODO doesn't seem to apply anymore based on its description, so
remove it altogether.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
scripts: compliance: Make line optional in fmtd_failure()

Some errors are attributable to a specific file, but not a specific line
within that file. Allow for formatted errors that refer to a file but
not a line.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
scripts: compliance: Clean up ZEPHYR_BASE usage

The global variable ZEPHYR_BASE is always set in the script, so no need
to check for its existence.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
scripts: compliance: Resolve ZEPHYR_BASE in _main()

Instead of globally resolving ZEPHYR_BASE, move it to _main() so we can
catch potential exceptions when doing so and for consistency with the
rest of globals, like GIT_TOP. This also includes documenting a new
"<zephyr-base>" path hint for classes.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

gmarull commented on scripts/ci/check_compliance.py:
```diff
@@ -82,9 +82,10 @@ def __init__(self, severity, title, file, line, col=None, desc=""):
 
-        txt = f'\n{title}{description}\nFile:{file}\nLine:{line}' + \
+        txt = f'\n{title}{description}\nFile:{file}' + \
+              (f'\nLine:{line}' if line else '') + \
```
suggestion: `line or ‘’`

carlescufi replied:
No, actually that won't work because the string is `\nLine:{line}`

carlescufi committed:
scripts: compliance: Move Kconfig-related items to the class

Both get_defined_syms and UNDEF_KCONFIG_WHITELIST are used exclusively
by the Kconfig check, so move them to the class.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@gmarull couldn't apply your suggestion but added an extra commit

stephanosio commented on scripts/ci/check_compliance.py:
```diff
@@ -566,98 +579,81 @@ def check_no_undef_outside_kconfig(self, kconf):
+        "APP_LOG_LEVEL", # Application log level is not detected correctly as
+                         # the option is defined using a template, so it can't
+                         # be grepped
+        "ARMCLANG_STD_LIBC",  # The ARMCLANG_STD_LIBC is defined in the 
```
> scripts/ci/check_compliance.py:590 Trailing whitespace (trailing-whitespace)

check_compliance seems to be very effective against itself :)

carlescufi replied:
🤣

nashif commented:
post merge +1 :)
